### ISSUE TITLE
Test with VB/F# to make sure expression tree differences are handled

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#14572]
This is in particular reference to removal of Relinq since it was doing some normalization in this area.

- [x] VB string equality/comparison, #19681

**Comment by bricelam** [comment]
[These re-linq transformations](https://github.com/re-motion/Relinq/tree/develop/Core/Parsing/ExpressionVisitors/Transformation/PredefinedTransformations) handled things like this.

**Comment by ajcvickers** [comment]
Note from triage: add a cross-language functional tests project.

**Comment by bricelam** [comment]
I've added a VB test suite with coverage for things that we've fixed (string comparisons and checked arithmetic).

We still have a few known bugs:

- https://github.com/dotnet/efcore/issues/27202
- https://github.com/dotnet/efcore/issues/29185
- https://github.com/dotnet/efcore/issues/30773

But I think we can say that the VB part of this issue is done.

**Comment by gaglia** [comment]
Hi,

Is there any news regarding this problem?

We are testing with EF 8 and still encounter the same problems.
The problem is very impactful in our opinion, we have just finished a huge migration work from Entity Framework 4 to .NET 8 + EF 8 but we are experiencing big problems for certain queries due to these mistranslations for VB.NET

Thank you

**Comment by roji** [comment]
@gaglia this is a general issue tracking improving testing for VB/F# - please vote on specific translation issues so that we know what you're blocked by. See the issues linked to just above; if you're running into something that isn't already covered, please open a new issue with a minimal code sample that shows the problematic query.

**Comment by gaglia** [comment]
Hi @roji 

I'm experiencing the same problem already reported in issue #31578.

Wrong translation of WHERE with simple filters for string fields

If you need any other information, just let me know.
Thank you

**Comment by roji** [comment]
@gaglia then be sure to vote on #31578, which currently has just one vote - that's a big part of how we choose which issues to prioritize.

**Comment by koenigst** [comment]
I created a test project for F# in #34541. I don't know of any issues concerning difference in syntax trees with F# so I created a sample test. Are any issues known that I could test?

**Comment by roji** [comment]
@koenigst I don't think we're aware of any specific problems, but then again EF has generally received little usage under F#, so it may be that the issues haven't been discovered/reported yet...

**Comment by bricelam** [comment]
The [area-F#](https://github.com/dotnet/efcore/labels/area-F%23) label and [efcore/EFCore.FSharp](https://github.com/efcore/EFCore.FSharp/issues) repo might have a few known issues in them.

**Comment by koenigst** [comment]
@bricelam The `area-F#` label has no open or closed issues concerning query translation. I found [one](https://github.com/efcore/EFCore.FSharp/issues/113), [two](https://github.com/efcore/EFCore.FSharp/issues/92) issues in `efcore/EFCore.FSharp`. To test them the test project would need to take a dependency on that project which I would find weird. Take the dependency or leave it for now?

**Comment by roji** [comment]
Ah, interesting... So it seems that the EFCore.FSharp package has actual method translators to take care of some F# expression tree variance. I'd need to look (and probably discuss with the team), but I think we should consider bringing these into EF itself (we already have various normalization support for VB.NET constructs in the box). These also may make more sense as preprocessing normalization rather than a method translators.

Otherwise, for the time being, it seems that EFCore.FSharp is necessary for proper query translation of some constructs, so I think it makes sense to take a dependency on it from the F# functional test project - you can go ahead and do that.

**Comment by roji** [comment]
FYI opened #34552 to track moving the query translation stuff from EFCore.FSharp into EF.

**Comment by koenigst** [comment]
> Otherwise, for the time being, it seems that EFCore.FSharp is necessary for proper query translation of some constructs, so I think it makes sense to take a dependency on it from the F# functional test project - you can go ahead and do that.

The package [`EntityFrameworkCore.FSharp`](https://www.nuget.org/packages/EntityFrameworkCore.FSharp) does not exist in the `dnc` package sources. I don't have the rights to mirror the package from nuget.org.

I will mention the two solved issues in #34552 and not pursue them here.

**Comment by roji** [comment]
@koenigst makes sense, thanks for looking at it. The right thing to do is probably to bring in the query fixes from the external package into the core as per #34552.
